src/modules/simulation/gz_bridge/GZBridge.cpp

```diff
@@ -748,6 +757,33 @@ void GZBridge::navSatCallback(const gz::msgs::NavSat &nav_sat)
 	pthread_mutex_unlock(&_node_mutex);
 }
 
+
+void GZBridge::LaserScantoLidarSensorCallback(const gz::msgs::LaserScan &scan)
+{
+
+	if (hrt_absolute_time() == 0) {
+		return;
+	}
+
+	distance_sensor_s distance_sensor{};
+	distance_sensor.timestamp = hrt_absolute_time();
+	distance_sensor.device_id = 0;
+	distance_sensor.min_distance = static_cast<float>(scan.range_min());
+	distance_sensor.max_distance = static_cast<float>(scan.range_max());
+	distance_sensor.current_distance = static_cast<float>(scan.ranges()[0]);
+	distance_sensor.variance = 0.0f;
+	distance_sensor.signal_quality = -1;
+	distance_sensor.type = distance_sensor_s::MAV_DISTANCE_SENSOR_LASER;
+	distance_sensor.h_fov = static_cast<float>(scan.angle_max() - scan.angle_min());
+	distance_sensor.v_fov = static_cast<float>(scan.angle_max() - scan.angle_min());
+	distance_sensor.q[0] = scan.world_pose().orientation().x();
+	distance_sensor.q[1] = scan.world_pose().orientation().y();
+	distance_sensor.q[2] = scan.world_pose().orientation().z();
+	distance_sensor.q[3] = scan.world_pose().orientation().w();
```

Comment:
This is probably not the best way to assemble the distance sensor topic, as it is using the ground truth orientation. How does the original distance sensor driver work?